core/src/main/java/nu/yona/server/subscriptions/service/BuddyDisconnectMessageDTO.java

```diff
@@ -106,7 +106,15 @@ private MessageActionDTO handleAction_Process(UserDTO actingUser, BuddyDisconnec
 		{
 			buddyService.removeBuddyAfterBuddyRemovedConnection(actingUser.getID(), messageEntity.getUserID());
 
+			updateMessageStatusAsProcessed(messageEntity);
```

Comment:
Nice catch! By the way, in https://github.com/yonadev/yona-server/pull/65 I
noticed that the status was not exposed on the message DTO at all... That
means that the status of no message was checked after doing some action on
it.

An idea: can we not embed the updated message DTO inside the
MessageActionDTO response? That will be more efficient for the app to
update existing messages.

On Mon, Dec 14, 2015 at 9:47 PM, Bert Roos <notifications@github.com> wrote:

> In
> core/src/main/java/nu/yona/server/subscriptions/service/BuddyDisconnectMessageDTO.java
> <https://github.com/yonadev/yona-server/pull/67#discussion_r47555281>:
>
> > @@ -106,7 +106,15 @@ private MessageActionDTO handleAction_Process(UserDTO actingUser, BuddyDisconnec
> >  		{
> >  			buddyService.removeBuddyAfterBuddyRemovedConnection(actingUser.getID(), messageEntity.getUserID());
> >
> > +			updateMessageStatusAsProcessed(messageEntity);
>
> This fixes a bug: the message status didn't flip to "processed" after
> processing it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/yonadev/yona-server/pull/67/files#r47555281>.
>